Update adding-language-variants.md

jonathoncove2 opened the pull request:
Added some support for setting cultures in API calls

## Description

_What did you add/update/change?_

## Type of suggestion

* [ ] Typo/grammar fix
* [ ] Updated outdated content
* [x] New content
* [ ] Updates related to a new version
* [ ] Other

## Product & version (if relevant)



## Deadline (if relevant)

_When should the content be published?_

eshanrnh commented:
Thanks for the PR, @jonathoncove2 🙌 We will review it as soon as we can. 💪

sofietoft commented:
Thanks for the PR @jonathoncove2 ! 💪 

The "Creating a basic site" tutorial is meant to be very high-level and shouldn't dive into more code than the cshtml and CSS files.
Instead, a fit for this code of yours, could be in the "[Creating a multilingual site](https://docs.umbraco.com/umbraco-cms/tutorials/multilanguage-setup)" instead?
Let me know what you think.
I'll make sure we have the code reviewed in the meantime 💪

jonathoncove2 commented:
Thanks for looking over this @sofietoft ! I agree that page is a more suitable place for this code. Would you prefer me to make a new PR for this? 

I also considered some example code showing how to add a language switching menu, would that be a useful addition?

sofietoft commented:
Excellent!
No need for a new PR, I think. Just put the code in the other article instead, then we're golden ⭐ 

It would be amazing with a language switcher 👏 
We've actually had that on an internal to-do list for ages, just never had the time to prioritize it.

The "Create a multilingual site" tutorial is a bit long and messy at the moment.
But if you could add your stuff where you think it makes sense, then I'll make sure we give it a look afterward - might need a bit of restructuring and perhaps it could even be split into multiple sub-articles 🤔

jonathoncove2 commented:
I've removed that API example as requested and added a language switcher. Not sure if my link from adding-language-variants.md works correctly, that might need to be fixed. I've also added sub headers, hopefully to guide users googling the same things I was. 

I'm about to go on holiday for a week and half, but feel free to Umbraco-ify the PR and approve it while I'm away.

sofietoft commented:
Amazing @jonathoncove2 !
I'll make sure we take a look at it as soon as possible 💪 

Have a wonderful holiday 🏖️

Frost117 reviewed (commented):
@jonathoncove2 I made some comments about the ILocalizationService, do you think you can rewrite the part but using the ILanguage service instead?

sofietoft commented:
Hello @jonathoncove2 👋 

Did you notice the review on your PR here? 😄

sofietoft commented:
Perfect, thanks @jonathoncove2 ! 💪 

I'll make sure our developer give this a second review.

Frost117 commented on 10/umbraco-cms/tutorials/multilanguage-setup.md:
```diff
@@ -226,3 +226,105 @@ For viewing purposes, I've added a stylesheet to my website. The final result sh
+    }
+
+    public IActionResult Index(string culture = null)
+    {
```
Hi @jonathoncove2 

I found the LanguageServiceTests.cs in the v14/dev repository and it does a Async call to get all the languages:
https://github.com/umbraco/Umbraco-CMS/blob/e4dbf4ce9ff9ca70ed7c82c0c735a4d576aad692/tests/Umbraco.Tests.Integration/Umbraco.Infrastructure/Services/LanguageServiceTests.cs#L15

Could you try and update this controller to inject the ILanguageService here and then use the LanguageService.GetAllAsync() function and see if it provides the same result as the localizationService?

Frost117 reviewed (approved):
Looks good with the v14 and the updated LanguageService 😃

sofietoft commented on 14/umbraco-cms/tutorials/multilanguage-setup.md:
```diff
@@ -262,3 +262,107 @@ Danish Version:
+
+You need to replace `{{homeNodeContentAlias}}` with the Document Type alias of your Home node. 
+
+This will look at all the cultures available on the home node, and render links to either the language variant of the current page or the home node for the language variant. If the home node for a language variant is removed, it will not appear in the list.
```
A suggestion to shorten this sentence a bit.

```suggestion
This will look at all the cultures available on the home node. Links to either the language variant of the current page or the home node for the language variant will be rendered. If the home node for a language variant is removed, it will not appear in the list.
```

sofietoft reviewed (commented):
I've taken the liberty to fix some grammar issues and some typos.
I've also fixed a couple of lists, where each item didn't start with the action word, which prefer 🙌 

I was wondering if you could put the steps in the **Steps** section into an ordered list? Since they are steps, I think it would be nice to format them as such as well 😄 

Finally, I've made a suggestion!
I hope it all makes sense. Let me know if something doesn't 💪

jonathoncove2 commented:
Thanks Sophie! Looks like I rely on spell checkers too much 🙈 

I've made the changes as you requested

sofietoft commented:
Looks great @jonathoncove2 ! 😄 
Think for applying the suggestions.

I'lle get this merged as soon as the check clears 🙌

jonathoncove2 commented:
Thanks @sofietoft !
